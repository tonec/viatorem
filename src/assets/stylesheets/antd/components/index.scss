@import "./affix/index";
@import "./alert/index";
@import "./anchor/index";
@import "./auto-complete/index";
@import "./avatar/index";
@import "./back-top/index";
@import "./badge/index";
@import "./breadcrumb/index";
@import "./button/index";
@import "./calendar/index";
@import "./card/index";
@import "./carousel/index";
@import "./cascader/index";
@import "./checkbox/index";
@import "./collapse/index";
@import "./date-picker/index";
@import "./divider/index";
@import "./dropdown/index";
@import "./form/index";
@import "./grid/index";
@import "./input/index";
@import "./input-number/index";
@import "./layout/index";
@import "./list/index";
@import "./mention/index";
@import "./menu/index";
@import "./message/index";
@import "./modal/index";
@import "./notification/index";
@import "./pagination/index";
@import "./popover/index";
@import "./progress/index";
@import "./radio/index";
@import "./rate/index";
@import "./select/index";
@import "./slider/index";
@import "./spin/index";
@import "./steps/index";
@import "./switch/index";
@import "./table/index";
@import "./tabs/index";
@import "./tag/index";
@import "./time-picker/index";
@import "./timeline/index";
@import "./tooltip/index";
@import "./transfer/index";
@import "./tree/index";
@import "./tree-select/index";
@import "./upload/index";