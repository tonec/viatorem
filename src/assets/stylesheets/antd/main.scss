@import "./themes/default";
@import "./core/index";
@import "./components/index";

div#root {
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  background: #F0F2F5;

  > div {
    width: 100%;
  }
}
