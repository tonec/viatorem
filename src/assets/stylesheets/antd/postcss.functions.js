const tinyColor = require('./functions/tinyColor.js')
const bezierEasing = require('./functions/bezierEasing.js')
const colorPalette = require('./functions/colorPalette.js')

module.exports = { tinyColor, bezierEasing, colorPalette }
