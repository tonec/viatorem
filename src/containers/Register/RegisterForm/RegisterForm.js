import React, { Component } from 'react'
import { reduxForm, Field, propTypes } from 'redux-form'
import { Form, Button } from 'antd'
import { TextField } from 'components/Forms/Inputs'
import registerValidation from './registerValidation'

@reduxForm({
  form: 'register',
  validate: registerValidation
})
export default class RegisterForm extends Component {

  static propTypes = {
    ...propTypes
  }

  render () {
    const { handleSubmit, error } = this.props

    return (
      <Form layout="inline" onSubmit={handleSubmit}>
        <Field
          name="name"
          type="text"
          component={TextField}
          label="Name"
        />
        <Field
          name="email"
          type="text"
          component={TextField}
          label="Email"
        />
        <Field
          name="password"
          type="password"
          component={TextField}
          label="Password"
        />
        <Field
          name="password_confirmation"
          type="password"
          component={TextField}
          label="Password confirmation"
        />
        <Form.Item>
          <Button type="primary" htmlType="submit">
            Log In
          </Button>
        </Form.Item>
        {error && <p className="help is-danger">{error}</p>}
      </Form>
    )
  }
}
