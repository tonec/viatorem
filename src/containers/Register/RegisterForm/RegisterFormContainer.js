import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { compose } from 'redux'
import { connect } from 'react-redux'
import { reduxForm } from 'redux-form'
import { register } from 'redux/modules/auth/actions'
import validation from './validation'
import RegisterForm from './RegisterForm'

export class RegisterFormContainer extends Component {

  static propTypes = {
    register: PropTypes.func.isRequired
  }

  render () {
    return (
      <RegisterForm {...this.props} onSubmit={this.props.register} />
    )
  }
}

const mapStateToProps = () => ({})

export default compose(
  connect(mapStateToProps, { register }),
  reduxForm({
    form: 'register',
    validate: validation
  })
)(RegisterFormContainer)
