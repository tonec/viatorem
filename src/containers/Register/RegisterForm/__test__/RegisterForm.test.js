import React from 'react'
import { shallow } from 'enzyme'
import { Alert } from 'antd'
import RegisterForm from '../RegisterForm'

describe('RegisterForm', () => {

  const props = {
    handleSubmit: jest.fn(),
    onSubmit: jest.fn(),
    submitting: false,
    error: null
  }

  it('renders according to snapshot', () => {
    const component = shallow(<RegisterForm {...props} />)
    expect(component).toMatchSnapshot()
  })

  it('display the error response when submission fails', () => {
    const error = 'Authentication failed'
    const component = shallow(<RegisterForm {...props} error={error} />)
    expect(component.find(Alert)).toHaveLength(1)
  })
})
