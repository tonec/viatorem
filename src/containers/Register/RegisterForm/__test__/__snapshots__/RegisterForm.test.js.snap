// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm renders according to snapshot 1`] = `
<Form
  hideRequiredMark={false}
  layout="vertical"
  onSubmit={[Function]}
  prefixCls="ant-form"
>
  <Field
    component={[Function]}
    name="name"
    placeholder="Name"
    size="large"
    type="text"
  />
  <Field
    component={[Function]}
    name="email"
    placeholder="Email"
    size="large"
    type="text"
  />
  <Field
    component={[Function]}
    name="password"
    placeholder="Password"
    size="large"
    type="password"
  />
  <Field
    component={[Function]}
    name="password_confirmation"
    placeholder="Password confirmation"
    size="large"
    type="password"
  />
  <FormItem
    colon={true}
    hasFeedback={false}
    prefixCls="ant-form"
  >
    <Button
      ghost={false}
      htmlType="submit"
      loading={false}
      prefixCls="ant-btn"
      size="large"
      type="primary"
    >
      Log In
    </Button>
  </FormItem>
</Form>
`;
